cli/src/semgrep/error_handler.py

```diff
@@ -0,0 +1,78 @@
+import sys
+from typing import Any
+from typing import Dict
+from typing import Optional
+
+import requests
+from attr import define
+from attr import Factory
+from attr import field
+
+from semgrep.verbose_logging import getLogger
+
+logger = getLogger(__name__)
+
+
+@define
+class ErrorHandler:
+    """
+    Send scan status to the fail-open endpoint
+    """
+
+    suppress_errors: bool = field(default=False)
+    payload: Dict[str, Any] = Factory(dict)
+
+    def configure(self, suppress_errors: Optional[bool] = False) -> None:
+        """
+        Configures whether to always or never send fail-open status.
+
+        :param suppress_errors: The value of the --suppress-errors option
+        """
+        if suppress_errors:
+            self.suppress_errors = suppress_errors
+
+    def push_request(self, method: str, url: str, **kwargs: Any) -> None:
+        self.payload = {"method": method, "url": url, **kwargs}
+
+    def append_request(self, **kwargs: Any) -> None:
+        self.payload = {**self.payload, **kwargs}
+
+    def pop_request(self) -> None:
+        self.payload = {}
+
+    @property
+    def is_enabled(self) -> bool:
+        """
+        Returns whether scan status should be sent.
+        """
+        return self.suppress_errors
+
+    def send(self, exit_code: int) -> int:
+        """
+        Send scan status to the fail-open server.
+
+        Will if is_enabled is True
```

Comment:
Missing info on what this will do if enabled :D

Reply:
Oops, copy/pasted from metrics.py, which is also lacking a description. Updating docstring in latest commit